sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java

```diff
@@ -0,0 +1,367 @@
+     * @param args Arguments for the message. If an exception is being logged, the last argument should be the
+     *     {@link Throwable}.
+     */
+    @Override
+    public void info(String format, Object... args) {
+        logFromFormat(LogLevel.INFORMATIONAL, format, args);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param format The formattable message to log.
+     * @param args Arguments for the message. If an exception is being logged, the last argument should be the
+     *     {@link Throwable}.
+     */
+    @Override
+    public void warn(String format, Object... args) {
+        logFromFormat(LogLevel.WARNING, format, args);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param format The formattable message to log.
+     * @param args Arguments for the message. If an exception is being logged, the last argument should be the
+     *     {@link Throwable}.
+     */
+    @Override
+    public void error(String format, Object... args) {
+        logFromFormat(LogLevel.ERROR, format, args);
+    }
+
+    /**
+     * Format and write the message according to the {@code MESSAGE_TEMPLATE}.
+     *
+     * @param level The level to log.
+     * @param format The log message format.
+     * @param arguments a list of arbitrary arguments taken in by format.
+     */
+    private void logFromFormat(LogLevel level, String format, Object... arguments) {
+        FormattingTuple tp = MessageFormatter.arrayFormat(format, arguments);
+        log(level, tp.getMessage(), tp.getThrowable());
+    }
+
+    /**
+     * Format and write the message according to the {@code MESSAGE_TEMPLATE}.
+     *
+     * @param level log level
+     * @param message The message itself
+     * @param t The exception whose stack trace should be logged
+     */
+    private void log(LogLevel level, String message, Throwable t) {
+        String dateTime = getFormattedDate();
+        String threadName = Thread.currentThread().getName();
+        String levelName = level.name();
+        StringBuilder buf = new StringBuilder(32);
+        buf.append(String.format(MESSAGE_TEMPLATE, dateTime, threadName, levelName, classPath, message));
+        writeWithThrowable(buf, t);
+    }
+
+    /**
+     * Get the current time in Local time zone.
+     *
+     * @return The current time in {@code DATE_FORMAT}
+     */
+    private String getFormattedDate() {
+        LocalDateTime now = LocalDateTime.now();
+        return DATE_FORMAT.format(now);
+    }
+
+    /**
+     * Write the log message with throwable stack trace if any.
+     *
+     * @param buf Take the log messages.
+     * @param t The exception whose stack trace should be logged
+     */
+    void writeWithThrowable(StringBuilder buf, Throwable t) {
+        if (t != null) {
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            t.printStackTrace(pw);
+            buf.append(sw.toString());
+            pw.close();
```

Comment:
Use `try-with-resources` to ensure the resource is always closed.

Reply:
Since StringWriter has no-op close(), I only move pw to try with resource. Thanks for pointing this out!